README.md

```diff
@@ -32,22 +33,237 @@ scheduled_task { 'Run Notepad':
+
+* Default for `operatingsystem` == `windows`.
+
+##### `trigger`
+
+One or more triggers defining when the task should run.
+A single trigger is represented as a hash, and multiple triggers can be specified with an array of hashes.
+
+A trigger can contain the following keys:
+
+For all triggers:
+
+* `schedule` (Required) — What kind of trigger this is.
+  Valid values are `daily`, `weekly`, `monthly`, or `once`.
+  Each kind of trigger is configured with a different set of keys; see the sections below (once triggers only need a start time/date.)
+* `start_time` (Required) — The time of day when the trigger should first become active.
+  Several time formats will work, but we suggest 24-hour time formatted as HH:MM.
+* `start_date` — The date when the trigger should first become active.
+  Defaults to the current date.
+  You should format dates as YYYY-MM-DD, although other date formats may work (under the hood, this uses Date.parse).
+* `minutes_interval` — The repeat interval in minutes.
+* `minutes_duration` — The duration in minutes, needs to be greater than the minutes_interval.
+* For daily triggers:
+  * `every` — How often the task should run, as a number of days.
+    Defaults to 1.
+    “2” means every other day, “3” means every three days, etc.
+* For weekly triggers:
+  * `every` — How often the task should run, as a number of weeks.
+    Defaults to 1.
+    “2” means every other week, “3” means every three weeks, etc.
+  * `day_of_week` — Which days of the week the task should run, as an array.
+    Defaults to all days.
+    Each day must be one of `mon`, `tues`, `wed`, `thurs`, `fri`, `sat`, `sun`, or `all`.
+* For monthly (by date) triggers:
+  * `months` — Which months the task should run, as an array.
+    Defaults to all months.
+    Each month must be an integer between 1 and 12.
+  * `on` (Required) — Which days of the month the task should run, as an array.
+    Each day must be an integer between 1 and 31.
+* For monthly (by weekday) triggers:
+  * `months` — Which months the task should run, as an array. Defaults to all months. Each month must be an integer between 1 and 12.
+  * `day_of_week` (Required) — Which day of the week the task should run, as an array with only one element.
+    Each day must be one of `mon`, `tues`, `wed`, `thurs`, `fri`, `sat`, `sun`, or `all`.
+  * `which_occurrence` (Required) — The occurrence of the chosen weekday when the task should run. Must be one of first, second, third, fourth, or fifth.
+
+##### `user`
+
+The user to run the scheduled task as.
+Please note that not all security configurations will allow running a scheduled task as 'SYSTEM', and saving the scheduled task under these conditions will fail with a reported error of 'The operation completed successfully'.
+It is recommended that you either choose another user to run the scheduled task, or alter the security policy to allow v1 scheduled tasks to run as the 'SYSTEM' account.
```

Comment:
`security policy` needs a link to the actual policy to be set.

Reply:
_Very_ good point, will add.

Reply:
Actually, my googlefu is failing me here, I can't find a specific security policy that blocks this. Still looking...

Reply:
It looks like this text was included in the [original type docs](https://github.com/puppetlabs/puppet/commit/0d7c79727cfb158f7182f791ca3ad6e6248c1c23).

Reply:
@joshcooper thinks it's related to [this article](https://support.microsoft.com/en-us/help/305612/how-to-prevent-a-user-from-running-task-scheduler-in-windows), possibly preventing Puppet from saving the task in the first place.

Reply:
Thinking more on this, we can't definitively find a policy that matches up with this text. What we discovered is a policy that, if intentionally set to prevent scheduled tasks from being added/saved, will prevent this resource from working. However, that seems like the policy functioning as intended and isn't the default on any systems.

I've removed this text from the documentation.

Reply:
Similar notes for this are in the comment I left at https://github.com/puppetlabs/puppetlabs-scheduled_task/pull/18#discussion_r171111673